types/rdf-js/index.d.ts

```diff
@@ -297,7 +297,7 @@ export interface DataFactory {
      * @see Triple
      * @see DefaultGraph
      */
-    triple<Q_In extends BaseQuad = Quad,  Q_Out extends BaseQuad = Quad>(subject: Q_In['subject'], predicate: Q_In['predicate'], object: Q_In['object']): Q_Out;
+    triple<Q extends BaseQuad = Quad>(subject: Q['subject'], predicate: Q['predicate'], object: Q['object']): Q;
```

Comment:
Can you check whether we have noUnusedGenerics disabled in the `tslint.json` for this package? If you care to work on them, there may be others in this package.

Reply:
Yes, if you remove that line entirely it should enable that rule.

Reply:
@LaurensRietveld Any update on this one?
I'd be happy to update it myself if you can give me access to your fork.

Reply:
I'm a bit confused ;). What is the problem and what is the proposed solution?

Is the problem having the `no-unnecessary-generics` flag set to false? If so, what is the recommended solution?
I think this situation results in a false-positive for this lint rule. If I see the doc for that rule, I see that it `Forbids a function to use a generic type parameter only once`. We're not using it once right? We're using if for 3 arguments and the return type.

Reply:
> If so, what is the recommended solution?

I think @DanielRosenwasser just means that `no-unnecessary-generics` should be set to true by removing the line from `tslint.json`.

> We're not using it once right? We're using if for 3 arguments and the return type.

Indeed, it should be fine here. But I guess there's nothing wrong in double-checking and making the compiler settings a bit more restrictive :-)